app/templating/view_context.py

```diff
@@ -53,13 +53,18 @@ def build_view_context_for_question(current_location, variables, rendered_block,
             'answer_errors': {},
             'data': {},
             'other_answer': {},
-            'fields': {}
+            'fields': {},
+            'question_titles': {}
```

Comment:
I don't think the `question_titles` should be nested inside the `form` object. They don't belong to it.

In the absence of a proper view model I'd suggest patching the title directly onto the `question` objects within each `rendered_block`.

```
for question in rendered_block.get('questions'):
    question['question_title`] = get_the_title()
```

Then the templates wouldn't need any logic for working out where to get the title from.

Reply:
We raised something similar as a possibility at the beginning but we shied away from that after sharing the ideas because of the need to know if the question was inside a repeating group / summary . We can do it but it would mean a significant rewrite .